docs/current/cookbook/snippets/cache-dependencies/main.py

```diff
@@ -0,0 +1,43 @@
+import sys
+
+import anyio
+
+import dagger
+
+
+async def main():
+    config = dagger.Config(log_output=sys.stdout)
+
+    async with dagger.Connection(config) as client:
+        # use a python:3.11 container
+        # mount the source code directory on the host
+        # at /src in the container
+        # mount the cache volumes to persist dependencies
+        source = (
+            client.container()
+            .from_("python:3.11")
+            .with_directory(
+                "/src",
+                client.host().directory("."),
+                exclude=[".venv/", ".cache/", "ci/"],
+            )
+            .with_workdir("/src")
+            .with_mounted_cache(
+                "/root/.cache/pip", client.cache_volume("pip-python-311-myapp-myenv")
+            )
+            .with_mounted_cache(
+                "/root/.local/pipx/cache",
+                client.cache_volume("pipx-python-311-myapp-myenv"),
+            )
+            .with_mounted_cache(
+                "/root/.cache/hatch",
+                client.cache_volume("hatch-python-311-myapp-myenv"),
+            )
```

Comment:
I know, but I wanted to show an example of how this could be done. Would it be better to comment it?

Reply:
I don't think that's necessary. Otherwise you'd be missing poetry, pipenv, etc. In node yarn as well. What's important is for people to understand how this caching works and to find the right cache paths for whatever tool they're using.

Reply:
Done